sparsehash/traits

```diff
@@ -52,4 +52,14 @@ struct is_relocatable<std::pair<T, U>>
 
 template <class T>
 struct is_relocatable<const T> : is_relocatable<T> {};
-}
\ No newline at end of file
+}
+
+#if defined _WIN32 || defined __CYGWIN__
```

Comment:
I spent a lot of time trying to remove all the ifdef's from the code :-)

Can we maybe look at just fixing hashtable_unittests.cc to not test those particular typedefs?
If not, this code belongs in the test file and not here :-)